### DISCUSSION TITLE
advice on disabling translation functionality?

### DISCUSSION THREAD

**Discussion by ClaireCJS** [zh-plus/openlrc#8] [New Feature]
Hi,

I am planning on running this on as many as 20,000 songs to create LRC files.  But 95% of my songs are in English.

So I want to skip the language translation functionality.

1) Because this may take months to run, any savings will help.  Even if translation is 10% of the time, that could save a week.

2) If I translate it all I may incur API fees about $250-400 by my calculations. 

3) And I'd be doing that to translate english to english, lol

Maybe I'm being paranoid about the money, but I definitely want the speed-up of skipping the translation.

Is there an option or some modification I can make to disable it?

**Comment by zh-plus** [comment]
I have added a new option called `skip_trans` in version 0.1.4 of the package, which is now available on PYPI.

To update the package from PYPI, you can use the following command:

```shell

pip install openlrc -U

```

After updating, you can try out by running this code:

```python

lrcer = LRCer()

lrcer.run('./data/test.mp3', target_lang='en', skip_trans=True)

```

Note that if you are transcribing Japanese or Chinese language, you may need to install additional `spacy` packages (breaking changes in v0.1.4). For more information and troubleshooting related to this issue, please refer to https://github.com/zh-plus/Open-Lyrics#installation.

**Comment by ClaireCJS** [comment]
Thank you, the option is working now!  You're amazing!

**Comment by ClaireCJS** [comment]
Could we have an option to skip the language detection part too?

Only because I am trying to speed things up so I can run this thousands of times :) And mine are always (95%) english :)

**Comment by ClaireCJS** [comment]
Small bug now - when skip_trans is set to True, the LRC file doesn't end up with the right name. 

Instead of "08_Ride On.mp3" creating "08_Ride On.lrc", with skip_trans=True, I get "08_Ride On_transcribed_optimized.lrc" 

![image](https://github.com/zh-plus/Open-Lyrics/assets/789591/98cd87b9-8b97-4e87-8f98-aec5ffac221c)
